### PR TITLE
Add support for closable tabs

### LINKED ISSUE #7498: Bokeh Panel(closable=True) doesn't make 'x' appear on the Panel title and closable by user
Hi, 
I am running your example for Panel and Tabs in bokeh. The 'closable=True' not working as expected.
Tested on Ubuntu 16.04, Chrome and Firefox browsers.
Using Bokeh 0.12.13 server and Python 3.6. 
Nothing special showed in the debugging.

Thanks

```
from bokeh.models.widgets import Panel, Tabs
from bokeh.io import output_file, show
from bokeh.plotting import figure
from bokeh.io import curdoc


output_file("slider.html")

p1 = figure(plot_width=300, plot_height=300)
p1.circle([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], size=20, color="navy", alpha=0.5)
tab1 = Panel(child=p1, title="circle")

p2 = figure(plot_width=300, plot_height=300)
p2.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], line_width=3, color="navy", alpha=0.5)
tab2 = Panel(child=p2, title="line", closable=True)

tabs = Tabs(tabs=[ tab1, tab2 ])

curdoc().add_root(tabs)
```

### PR DESCRIPTION
**Pull request by mattpap**
image

fixes #7498

### CODE REVIEW

**Comment by mattpap** [comment]
Currently if you remove a tab after scrolling headers, the view will jump to the beginning. I will fix this at some point under the umbrella of incremental DOM changes.

**Comment by mattpap** [comment]
Or I just scroll active header into view, which would be useful in general.

**Comment by bryevdv** [comment]
I don't have strong opinions about this feature @philippjfr @mattpap you two should decide when this has the most desirable behavior and is ready to merge.

File: bokeh/models/layouts.py [hunk 1]
```diff
@@ -412,6 +412,10 @@ class Panel(Model):
     The child widget. If you need more children, use a layout widget, e.g. a ``Column``.
     """)
 
+    closable = Bool(False, help="""
+    Whether this panel is closeable or not. If True, an "x" button will appear.
+    """)
```
Comment (bryevdv) [inline bokeh/models/layouts.py]: We should be explicit here that the value of `tabs` will be mutated, and the panel removed

**Comment by philippjfr** [comment]
This is looking good to me, I do agree that the docstring and docs should be very explicit that closing a tab will modify the tabs property.

**Comment by bryevdv** [comment]
@mattpap do you plan to add the docs, or do you want someone else to pick it up?

**Pull request merged**